src/subtype.c

```diff
@@ -1313,6 +1318,15 @@ static int subtype(jl_value_t *x, jl_value_t *y, jl_stenv_t *e, int param)
             e->invdepth = saved;
             return issub;
         }
+        if (jl_is_vararg_marker(xd)) {
+            if (!jl_is_vararg_marker(yd))
+                return 0;
+            // N.B.: This case is only used for raw varargs that are not part
+            // of a tuple (those that are have special handling in subtype_tuple).
+            // Vararg isn't really a proper type, but it does sometimes show up
+            // as e.g. Type{Vararg}, so we'd like to handle that correctly.
```

Comment:
`Type{Vararg}` is not allowed. Ideally we would give an error here too, but if that's too breaking we can instead just change this comment to say it's provided for compatibility with code that uses `<:` to compare Vararg objects.

Reply:
This comment is old, it just moved from below. I can try giving an error and see what needs to be updated.

Reply:
Oh, as Martin noticed above it looks like passing Vararg to `<:` is already an error in this PR :) We can try to go with that, and change this to an assertion.